cvat/apps/engine/cache.py

```diff
@@ -356,7 +358,7 @@ def _to_data_with_mime(self, cache_item: _CacheItem) -> DataWithMime: ...
     def _to_data_with_mime(self, cache_item: Optional[_CacheItem]) -> Optional[DataWithMime]: ...
 
     def _to_data_with_mime(self, cache_item: Optional[_CacheItem]) -> Optional[DataWithMime]:
-        if not cache_item:
+        if not cache_item or not len(cache_item[0].getbuffer()):
```

Comment:
You will need to update all the places where empty results were not expected. I know at least 1 such place - `get_frame_context_images_chunk()`. The `_to_data_with_mime()` function annotations are also not correct now.

Reply:
But it actually looks like we should just expect empty result (empty `BytesIO`) in places, where empty cached items can be valid.

Reply:
> But it actually looks like we should just expect empty result (empty BytesIO) in places, where empty cached items can be valid.

Why? I still think that there should be a common approach

Reply:
we should probably verify that data is not empty in places where it is not acceptable

Reply:
A common approach to what? The cache returns exactly what was put into the cache, this is the current approach.

Reply:
Yes, but I thought that you meant [here](https://github.com/cvat-ai/cvat/pull/8890#discussion_r1900093238) not to update `_to_data_with_mime` to return None when cache_item[0] is empty. Am I missing something?

Reply:
> A common approach to what?

Update _to_data_with_mime logic and not to patch each function where _to_data_with_mime is called